crates/parcel_config/src/parcel_rc_config.rs

```diff
@@ -0,0 +1,781 @@
+        source,
+      })?;
+
+    self.process_config(&ParcelRcFile { path, contents })
+  }
+
+  fn resolve_extends(
+    &self,
+    config_path: &PathBuf,
+    extend: &String,
+  ) -> Result<PathBuf, ConfigError> {
+    let path = if extend.starts_with(".") {
+      config_path.parent().unwrap_or(config_path).join(extend)
+    } else {
+      self
+        .package_manager
+        .resolve(extend, config_path)
+        .map_err(|source| ConfigError::UnresolvedConfig {
+          config_type: String::from("extended config"),
+          from: PathBuf::from(config_path),
+          source: Box::new(source),
+          specifier: String::from(extend),
+        })?
+        .resolved
+    };
+
+    self
+      .fs
+      .canonicalize_base(path.clone())
+      .map_err(|source| ConfigError::UnresolvedConfig {
+        config_type: String::from("extended config"),
+        from: path,
+        source: Box::new(source),
+        specifier: String::from(extend),
+      })
+  }
+
+  fn process_config(
+    &self,
+    parcel_rc: &ParcelRcFile,
+  ) -> Result<(PartialParcelConfig, Vec<PathBuf>), ConfigError> {
+    let mut files = vec![parcel_rc.path.clone()];
+    let extends = parcel_rc.contents.extends.as_ref();
+    let extends = match extends {
+      None => Vec::new(),
+      Some(extends) => match extends {
+        Extends::One(ext) => vec![String::from(ext)],
+        Extends::Many(ext) => ext.to_vec(),
+      },
+    };
+
+    if extends.is_empty() {
+      return Ok((PartialParcelConfig::try_from(parcel_rc)?, files));
+    }
+
+    let mut merged_config: Option<PartialParcelConfig> = None;
+    for extend in extends {
+      let extended_file_path = self.resolve_extends(&parcel_rc.path, &extend)?;
+      let (extended_config, mut extended_file_paths) = self.load_config(extended_file_path)?;
+
+      merged_config = match merged_config {
+        None => Some(extended_config),
+        Some(config) => Some(PartialParcelConfig::merge(config, extended_config)),
+      };
+
+      files.append(&mut extended_file_paths);
+    }
+
+    let config = PartialParcelConfig::merge(
+      PartialParcelConfig::try_from(parcel_rc)?,
+      merged_config.unwrap(),
+    );
+
+    Ok((config, files))
+  }
+
+  pub fn load(
+    &self,
+    project_root: &PathBuf,
+    options: LoadConfigOptions<'a>,
+  ) -> Result<(ParcelConfig, Vec<PathBuf>), ConfigError> {
+    let resolve_from = self.resolve_from(project_root);
+    let mut config_path = match options.config {
+      Some(config) => self
+        .package_manager
+        .resolve(&config, &resolve_from)
+        .map(|r| r.resolved)
+        .map_err(|source| ConfigError::UnresolvedConfig {
+          config_type: String::from("config"),
+          from: resolve_from.clone(),
+          source: Box::new(source),
+          specifier: String::from(config),
+        }),
+      None => self.find_config(project_root, &resolve_from),
+    };
+
+    let mut _used_fallback = false;
```

Comment:
nit: unused variable; should remove

Reply:
It's in the old impl so I left it there, but feel it the logic can be improved later once that piece gets used. But I can remove it in the meantime